vault/audit.go

```diff
@@ -413,33 +473,33 @@ func (c *Core) setupAudits(ctx context.Context) error {
 		// ensure that it is reset after. This ensures that there will be no
 		// writes during the construction of the backend.
 		view.SetReadOnlyErr(logical.ErrSetupReadOnly)
-		c.postUnsealFuncs = append(c.postUnsealFuncs, func() {
-			view.SetReadOnlyErr(origViewReadOnlyErr)
-		})
+		if req.isInitial {
+			c.postUnsealFuncs = append(c.postUnsealFuncs, func() {
+				view.SetReadOnlyErr(origViewReadOnlyErr)
+			})
+		} else {
+			defer view.SetReadOnlyErr(origViewReadOnlyErr)
+		}
 
 		// Initialize the backend
-		backend, err := c.newAuditBackend(ctx, entry, view, entry.Options)
+		backend, err := c.newAuditBackend(req.ctx, entry, view, entry.Options)
 		if err != nil {
-			c.logger.Error("failed to create audit entry", "path", entry.Path, "error", err)
+			multiErr = multierror.Append(multiErr, err)
 			continue
 		}
 		if backend == nil {
-			c.logger.Error("created audit entry was nil", "path", entry.Path, "type", entry.Type)
+			multiErr = multierror.Append(multiErr, fmt.Errorf("nil audit backend of type %q returned from factory at path %q", entry.Type, entry.Path))
 			continue
 		}
 
-		// Mount the backend
-		broker.Register(entry.Path, backend, view, entry.Local)
-
-		successCount++
-	}
-
-	if len(c.audit.Entries) > 0 && successCount == 0 {
-		return errLoadAuditFailed
+		// Register the backend
+		c.auditBroker.Register(entry.Path, backend, view, entry.Local)
+		if c.logger.IsInfo() {
+			c.logger.Info("enabled audit backend", "path", entry.Path, "type", entry.Type)
+		}
```

Comment:
```suggestion
	  c.logger.Info("enabled audit backend", "path", entry.Path, "type", entry.Type)
```

Nonsensical pattern that probably warrants a separate issue.